src/components/Steps.js

```diff
@@ -165,8 +167,13 @@ export default class Steps extends React.Component {
       return false;
     }
 
+    const goToStepOpt = {};
+    const stepPayment = this.props.stepPayment || {};
     const prevStep = this.props.steps[currentStep.index - 1];
-    this.goToStep(this.buildStep(prevStep, currentStep.index - 1));
+    if (currentStep.isLastStep && stepPayment.isNew && (stepPayment.error || !stepPayment.data)) {
```

Comment:
Would be great to add code comments to explain in plain english what we're doing here.